### PR TITLE
librustc error_reporting.rs cleanup.

### PR DESCRIPTION
**Pull request by frewsxcv**
Read some code in librustc, mainly in error_reporting.rs, and cleaned up some things along the way. I recommend looking at each commit individually or looking at the [whitespace insensitive diff](https://github.com/rust-lang/rust/pull/39977/files?w=1).

### CODE REVIEW

**Comment by eddyb** [comment]
I'm not sure on the style changes. I know there's a style team but I can't figure out how to ping them.
cc @nrc @solson @nikomatsakis

**Comment by solson** [comment]
@eddyb My approach to style in the rust repo is generally "I'm not going to worry about minor inconsistencies now because we're going to automate it all with rustfmt later".

> but I can't figure out how to ping them

You can't ping us here becase we're under the `rust-lang-nursery` org. We would need to be moved over here.

File: src/librustc/traits/error_reporting.rs [hunk 1]
```diff
@@ -359,34 +359,9 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
     }
 
     fn report_similar_impl_candidates(&self,
-                                      trait_ref: ty::PolyTraitRef<'tcx>,
+                                      impl_candidates: Vec<ty::TraitRef<'tcx>>,
                                       err: &mut DiagnosticBuilder)
     {
-        let simp = fast_reject::simplify_type(self.tcx,
```
Comment (eddyb) [inline src/librustc/traits/error_reporting.rs]: Was all of this duplicated from `find_similar_impl_candidates`? 😆

**Review by eddyb — APPROVED** [review]
Commit-by-commit makes a lot more sense!

File: src/librustc/traits/error_reporting.rs (hunk 1 above)
Comment (frewsxcv) [inline src/librustc/traits/error_reporting.rs]: Yeeeppppppp

**Comment by alexcrichton** [comment]
@bors: retry

* network error

**Comment by alexcrichton** [comment]
@bors: retry

* network failure

**Comment by alexcrichton** [comment]
@bors: retry

* osx linker segfault #38878

**Comment by alexcrichton** [comment]
@bors: retry

* sccache failure which means it needs updating
https://github.com/rust-lang/rust/pull/40095

**Pull request merged**